File: DotNetOnion/Cells/CellCreateFast.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text;
using DotNetOnion.Helpers;

namespace DotNetOnion.Cells
{
    public class CellCreateFast : Cell
    {
        public override byte Command => 0x05;

        public byte[] X { get; set; }

        public override void Deserialize(BinaryReader reader)
        {
            X = reader.ReadBytes(Constants.HashLength);
        }

        public override void Serialize(BinaryWriter writer)
        {
            writer.Write(X);
        }
    }
}

File: NOnion/Cells/CellCreateFast.fs
namespace NOnion.Cells

open System.IO

open NOnion

type CellCreateFast =
    {
        X: array<byte>
    }

    static member Deserialize(reader: BinaryReader) =
        {
            X = reader.ReadBytes Constants.HashLength
        }
        :> ICell

    interface ICell with

        member __.Command = 5uy

        member self.Serialize writer =
            writer.Write self.X
